internal/controllers/device_data_controller.go

```diff
@@ -230,6 +217,7 @@ func (d *DeviceDataController) getHistoryV1(c *fiber.Ctx, userDevice *grpc.UserD
 		},
 		Size:    some.Int(1000),
 		Source_: &source,
+		Sort:    []types.SortCombinations{types.SortOptions{SortOptions: map[string]types.FieldSort{"data.timestamp": {Order: &sortorder.Asc}}}},
```

Comment:
I worry that the default sort, by score, is what's making the random selection work.